test/others/test_utils.py

```diff
@@ -1193,3 +1198,26 @@ def test_get_eval_run_results():
     first_result = node_results.iloc[0]
     assert first_result["exact_match"] == True
     assert first_result["answer"] == "This"
+
+
+def test_exponential_backoff():
+    # Test that the exponential backoff works as expected
+    # should raise exception, check the exception contains the correct message
+    with pytest.raises(Exception, match="retries \(2\)"):
+
+        @retry_with_exponential_backoff(backoff_in_seconds=1, max_retries=2)
+        def greet(name: str):
+            if random() < 0.999:
+                raise OpenAIRateLimitError("Too many requests")
+            print(f"Hello {name}")
+
+        greet("Hello John")
+
+    # this should not raise exception and should print "Hello John"
+    @retry_with_exponential_backoff(backoff_in_seconds=1, max_retries=1)
+    def greet2(name: str):
+        if random() < 0.0001:
+            raise OpenAIRateLimitError("Too many requests")
+        print(f"Hello {name}")
+
+    greet2("Hello John")
```

Comment:
I think for this second test we could remove the if statement and raise Error statement since we never want it to raise anyways right? And as it is right now we run this risk of the test failing since `ramdon()` could return exactly `0.0`.